modules/engine/src/model/model.ts

```diff
@@ -565,23 +578,25 @@ export class Model {
 
   _updatePipeline(): RenderPipeline {
     if (this._pipelineNeedsUpdate) {
+      const prevShaders = [] as Shader[];
       if (this.pipeline) {
         log.log(
           1,
           `Model ${this.id}: Recreating pipeline because "${this._pipelineNeedsUpdate}".`
         )();
+        prevShaders.push(this.pipeline.vs, this.pipeline.fs);
```

Comment:
fs can be null in transform feedback, do we need to do `prevShaders.filter(Boolean)`?